MAINT: NumPy 2.x cleanups

tylerjereddy opened the pull request:
* A batch of cleanups following the bump in minimum required NumPy/Python versions in gh-24119.

tylerjereddy committed:
MAINT: NumPy 2.x cleanups

* A batch of cleanups following the bump in minimum required
NumPy version in gh-24119.

rgommers reviewed (approved):
Nice, thanks @tylerjereddy. All changes here LGTM.

tylerjereddy committed:
MAINT: PR 24173 revisions

* Based on reviewer feedback, the `np_vecdot()` function
has been removed and uses of it have been replaced with the
native NumPy equivalent.

tylerjereddy commented on scipy/_lib/_util.py:
```diff
@@ -1231,12 +1215,7 @@ def np_vecdot(x1, x2, /, *, axis=-1):
-        # of course there are other fancy ways of doing this (e.g. `einsum`)
-        # but let's keep it simple since it's temporary
-        return np.sum(x1.conj() * x2, axis=axis)
+    return np.vecdot(x1, x2, axis=axis)
```
ok, I pushed in a commit to do that
